docs/state_machines.rst

```diff
@@ -0,0 +1,58 @@
+Raiden Guide on State Machines
+##############################
+.. toctree::
+   :maxdepth: 2
+
+Introduction
+============
+
+The current Raiden prototype chose to use hierarchical state machines to represent the internal state of the node. The state of the whole node is a tree that can be serialized on and recovered from the disk. Moreover, whenever the state changes, the change is triggered by a state change object, which can also be serialized on the disk. A snapshot of the state and the subsequent state change objects are enough to compute the current state of the node. That's how a node recovers from crashes.
+
+Kinds of State Machines
+=======================
+
+The state of the whole node is represented as a ``ChainState`` object. The ``ChainState`` object contains many other different kinds of objects. This section describes them in a bottom-to-top manner.
+
+This section is very early in construction with the simplest state machines.
+
+HashTimeLockState
+*****************
+
+A ``HashTimeLockState`` represents a hash time lock.
+
+Attributes of a HashTimeLockState
```

Comment:
I would skip documenting the attributes of each state or state change. Simply point them to the documentation's docstring. No reason to repeat information here.

Reply:
I've added comments to the source.  I cannot add a link in the spec right now until the comments are merged.

Reply:
Source of the raiden client? Where?

Reply:
Here https://github.com/raiden-network/raiden/pull/3177/files#diff-a6042df3a2b6bca66372154b6cceff89R1000 in this PR.